Zend/zend_vm_execute.skl

```diff
@@ -16,6 +16,15 @@ ZEND_API void {%EXECUTOR_NAME%}_ex(zend_execute_data *ex)
 	LOAD_OPLINE();
 	ZEND_VM_LOOP_INTERRUPT_CHECK();
 
+#ifdef ZEND_CHECK_STACK_LIMIT
+	if (zend_call_stack_overflowed(EG(stack_limit))) {
+		zend_call_stack_size_error();
+		EG(opline_before_exception) = NULL;
```

Comment:
I'm not too happy with this hack here. AFAIK all functions do have a final return or throw op (which obviously it not in any try block (as try blocks precede other blocks) nor having temps needing to be freed (being the last instruction)). Can we just point the EG(opline_before_exception) to &EX(func)->op_array.opcodes[EX(func)->op_array.last - 1] ?

Reply:
Agreed. I added this as a quick fix so that I could move forward, but I will revisit this part. This may end up being the right solution though. The thing is that `ZEND_HANDLE_EXCEPTION` assumes that `EG(opline_before_exception)` was executed, which is not the case here, and tries to cleanup after it.

Reply:
To be more precise, ZEND_HANDLE_EXCEPTION assumes that everything what the opcode at EG(opline_before_exception) consumes already has been cleaned up. ZEND_HANDLE_EXCEPTION only cares about garbage having a longer lifetime than the exception. On the last exception, there will be no such garbage.

Reply:
> AFAIK all functions do have a final return or throw op (which obviously is not in any try block (as try blocks precede their catch or finally blocks) nor having temps needing to be freed (being the last instruction)). Can we just point the EG(opline_before_exception) to &EX(func)->op_array.opcodes[EX(func)->op_array.last - 1] ?

I think that this would work, but I'm not happy about this solution.

`EG(opline_before_exception) = NULL;` seems to be the natural thing to do. No opline was executed before the exception, so it's natural to have `EG(opline_before_exception)` set to `NULL`.

Most usages of `EG(opline_before_exception)` already have non-null checks before de-referencing it (except in JIT code).